src/services/gridPageFetcher.ts

```diff
@@ -68,22 +77,28 @@ export class GridPageFetcher {
     // TODO: we should make sure that only the latest cache operation is run
     // TODO: there might be a race condition here with changing the cache
     // heads a few lines down
+    this.refreshCache();
+  }
+
+  public async refreshCache(): Promise<void> {
     this.audioCacheClient();
     this.audioCacheServer();
-
-    return this.subjectQueueBuffer.splice(0, requestedItems);
   }
 
   // during client caching, we do a GET request to the server for the
   // audio file. Therefore, requests that have already been client cached
   // have also already been server cached. We therefore, remove these
   // requests from the calculations
   private async audioCacheClient(): Promise<void> {
-    const models = this.subjectQueueBuffer.slice(0, this.clientCacheLength).filter((model) => !model.clientCached);
+    const models = this.subjectQueue
+      .slice(0, this.clientCacheLength)
```

Comment:
why always slicing from 0? shouldn't be viewHead through to viewHead + clientCacheLength?